main.c

```diff
@@ -115,6 +115,18 @@ static struct kanshi_mode *match_mode(struct kanshi_head *head,
 	return NULL;
 }
 
+static void execute_profile_commands(struct kanshi_profile *profile) {
+	struct kanshi_profile_command *command;
+	wl_list_for_each(command, &profile->commands, link) {
+		int res = system(command->command);
```

Comment:
Using `system` here will wait until the command has finished. This is different from what Sway does (executing them in parallel).

Blocking seems a little dangerous because it's not integrated with the Wayland event loop and could potentially lead to a crash because the Wayland socket becomes full. The alternative would be to manually `fork` and `execl`.